google/ggrc-core
# GGRC-6755/GGRC-6778/GGRC-6779/GGRC-6780/GGRC-6781/GGRC-6782/GGRC-6783 Mega Program BE

# Issue description

Currently a program is created separate for each product group. Few programs should be able to be combined for a common audit. All mapped to child programs objects should be also mapped to Mega program. It should be possible to map one mega program to another mega program.

# Steps to test the changes
Create Mega Program:

using FE(not in this PR): Map child-/parent program using tabs on Program page. 
using BE: Create Relationship(source=mega_program,destination=child_program)

After program-to-program mapping objects from child program should be mapped to parent program. New object-to-program  mappings to child program should be also mapped to parent programs.

Mapping should happen to next list of objects:       "Regulation", "Objective", "Control", "Clause", "Contract", "Policy", "Risk", "Standard", "Threat", "Requirement", "System", "Product", "Process", "Market", "DataAsset", "Facility", "OrgGroup", "Metric", "TechnologyEnvironment", "ProductGroup", "Project", "Vendor", "AccessGroup"

Programs could be mapped many-to-many in graph, with few generations of parents/children and cycles. If cycle occurred, all objects related to programs in cycle would be mapped to all programs.

# Solution description

Mega program is a program that have child programs. 
Child programs are mapped using Relationship(source=mega_program,destination=child_program)

Program have attribute _is_mega_ that show existence relationships to child programs

Query API have new operators _child_ / _parent_ that return one level of related programs

Revisions returned for FE Change Log populated with automapping content to help FE show additional info in Change Log

Relationship revisions are sorted before commiting to DB in next order - first original(by user) mapping, then auto mappings(by system).  Because all revisions are created_at the same second and Change Log potentially could show automapping before original mapping.

# Sanity checklist




# PR Review checklist
